web/src/user_card_popover.js

```diff
@@ -554,50 +581,54 @@ function toggle_sidebar_user_card_popover($target) {
         return;
     }
 
+    // For the sidebar user_card_popover, we should have an offset
+    // to render the popover outside the user sidebar.
+    const tippyOptions = {
+        getReferenceClientRect: () => ({
+            width: 0,
+            height: 0,
+            left: 0,
+            top: 0,
+        }),
+        placement: "top",
+        popperOptions: {
+            modifiers: [
+                {
+                    name: "offset",
+                    options: {
+                        offset({popper}) {
+                            // Calculate the offset needed to place the popover in the correct place near sidebar
+                            const sidebar_width =
+                                window.innerWidth < media_breakpoints_num.xl ? 265 : 250;
+
+                            const x_offset_to_center =
+                                window.innerWidth - sidebar_width - popper.width / 2;
+                            const y_offset_to_center = 51;
```

Comment:
Can you extend the comments to explain where the 51/265/250 come from? If they are numbers that also exist in the CSS, we can make them CSS variables and then have this function read those CSS variables so that the values stay in sync.

Reply:
This also seems to not work consistently; at least changing browser zoom results in outcomes like this that don't seem intended. 

![image](https://github.com/zulip/zulip/assets/2746074/f5f50f74-c117-4815-8b7a-ecf7f8fe6f20)

Reply:
Clarifying with @timabbott , the problem in the above sceenshot is that this user card was triggered from the right sidebar and not placed correctly.

Reply:
> Can you extend the comments to explain where the 51/265/250 come from?

I calculated these values myself to get the popover in the right spot. But a better approach might be to base them on the vdots of the first user in the list, for example. That way, it'd work better with zooming and stuff.

For now, I think we can ditch this placement logic and let Tippy handle it, like we talked about on CZO [here](https://chat.zulip.org/#narrow/stream/6-frontend/topic/sidebar.20user.20card.20placement)

Reply:
Yeah, this solution seems to work well, resolving.